src/graph/util/ExpressionUtils.cpp

```diff
@@ -177,6 +180,39 @@ Expression *ExpressionUtils::rewriteParameter(const Expression *expr, QueryConte
   return graph::RewriteVisitor::transform(expr, matcher, rewriter);
 }
 
+Expression *ExpressionUtils::rewriteInnerInExpr(const Expression *expr) {
+  auto matcher = [](const Expression *e) -> bool {
+    if (e->kind() != Expression::Kind::kRelIn) {
+      return false;
+    }
+    auto rhs = static_cast<const RelationalExpression *>(e)->right();
+    return rhs->kind() == Expression::Kind::kList || rhs->kind() == Expression::Kind::kSet;
```

Comment:
Maybe should limit size of List to avoid stack overflow.

Reply:
Where is the stack overflow?

Reply:
@Shylock-Hg Construct flatten LogicalExpression instead.